### PR TITLE
Added enableMulti option to ButtonGroupInput

### PR DESCRIPTION
**Pull request by Ryanseanlee**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [ ] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [ ] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: desktop/cmp/input/ButtonGroupInput.js [hunk 1]
```diff
@@ -64,6 +78,23 @@ class Model extends HoistInputModel {
         const btns = this.domEl?.querySelectorAll('button') ?? [];
         return filter(btns, {disabled: false});
     }
+
+    active(value){
+        if (!this.enableMulti) return this.renderValue === value;
```
Comment (lbwexler) [inline desktop/cmp/input/ButtonGroupInput.js:83]: probably best to use "internalValue" to be consistent with usage below. (although will be the same for this control.  Also destructure `internalValue` and consider ternary!

File: desktop/cmp/input/ButtonGroupInput.js [hunk 2]
```diff
@@ -64,6 +78,23 @@ class Model extends HoistInputModel {
...
+    onButtonClick(value){
+        if (this.enableMulti){
+            const current = this.internalValue ?? [];
+            value = this.active(value) ?
+                current.length === 1 ? null : without(current, value) :
+                [...current, value];
+        } else {
+            value = this.enableClear && this.active(value) ? null : value;
+        }
+        this.noteValueChange(value);
```
Comment (lbwexler) [inline desktop/cmp/input/ButtonGroupInput.js:96]: 👍🏻   This logic looks great now.

**Comment by Ryanseanlee** [comment]
We have a problem initializing a value using the forms/ form model. When enableMulti is true we expect (other than null) an array of strings as the value of the button group. However, it may be common for the user to set the initial value of the form model to a string as that is how it works without enableMulti. We have no way of intercepting/ fixing this value because it is set directly and not set through one of the HoistInputModel functions.

https://github.com/xh/hoist-react/blob/998a64a555f37e417029bb62603b0f5c44548642/cmp/form/field/BaseFieldModel.js#L186-L190

File: mobile/cmp/input/ButtonGroupInput.js [hunk 3]
```diff
@@ -25,14 +26,20 @@ export const [ButtonGroupInput, buttonGroupInput] = hoistCmp.withFactory({
...
     }
 });
 ButtonGroupInput.propTypes = {
     ...HoistInputPropTypes,
 
     /** True to allow buttons to be unselected (aka inactivated). Defaults to false. */
-    enableClear: PT.bool
+    enableClear: PT.bool,
+
+    enableMulti: PT.bool
```
Comment (lbwexler) [inline mobile/cmp/input/ButtonGroupInput.js:42]: need jsdoc here.  Make both mirror desktop at the same time

**Comment by lbwexler** [comment]
This looks good Ryan.  Will cleanup the one docstring and merge.

Regarding the issue about a user setting the value type to something incorrect -- will move that text to another ticket -- think its a much more general bug that afflicts all of our controls.

**Pull request merged**